crates/consensus/src/transaction/eip4844.rs

```diff
@@ -857,73 +760,10 @@ impl TxEip4844WithSidecar {
     }
 
-    /// Encodes the transaction from RLP bytes, including the signature. This __does not__ encode a
-    /// tx type byte or string header.
-    ///
-    /// This __does__ encode a list header and include a signature.
-    ///
-    /// This encodes the following:
-    /// `rlp([tx_payload, blobs, commitments, proofs])`
-    ///
-    /// where `tx_payload` is the RLP encoding of the [TxEip4844] transaction fields:
-    /// `rlp([chain_id, nonce, max_priority_fee_per_gas, ..., v, r, s])`
-    pub fn encode_with_signature_fields<S>(&self, signature: &S, out: &mut dyn BufMut)
-    where
-        S: EncodableSignature,
-    {
-        let inner_payload_length = self.tx.fields_len() + signature.rlp_vrs_len();
-        let inner_header = Header { list: true, payload_length: inner_payload_length };
-
-        let outer_payload_length =
-            inner_header.length() + inner_payload_length + self.sidecar.rlp_encoded_fields_length();
-        let outer_header = Header { list: true, payload_length: outer_payload_length };
-
-        // write the two headers
-        outer_header.encode(out);
-        inner_header.encode(out);
-
-        // now write the fields
-        self.tx.encode_fields(out);
-        signature.write_rlp_vrs(out);
-        self.sidecar.rlp_encode_fields(out);
-    }
-
-    /// Decodes the transaction from RLP bytes, including the signature.
-    ///
-    /// This __does not__ expect the bytes to start with a transaction type byte or string
-    /// header.
-    ///
-    /// This __does__ expect the bytes to start with a list header and include a signature.
-    ///
-    /// This is the inverse of [TxEip4844WithSidecar::encode_with_signature_fields].
-    #[doc(hidden)]
-    pub fn decode_signed_fields(buf: &mut &[u8]) -> alloy_rlp::Result<Signed<Self>> {
-        let header = Header::decode(buf)?;
-        if !header.list {
-            return Err(alloy_rlp::Error::UnexpectedString);
-        }
-
-        // record original length so we can check encoding
-        let original_len = buf.len();
-
-        // decode the inner tx
-        let inner_tx = TxEip4844::decode_signed_fields(buf)?;
-
-        // decode the sidecar
-        let sidecar = BlobTransactionSidecar::rlp_decode_fields(buf)?;
-
-        if buf.len() + header.payload_length != original_len {
-            return Err(alloy_rlp::Error::ListLengthMismatch {
-                expected: header.payload_length,
-                got: original_len - buf.len(),
-            });
-        }
-
-        let (tx, signature, hash) = inner_tx.into_parts();
-
-        // create unchecked signed tx because these checks should have happened during construction
-        // of `Signed<TxEip4844>` in `TxEip4844::decode_signed_fields`
-        Ok(Signed::new_unchecked(Self::from_tx_and_sidecar(tx, sidecar), signature, hash))
+    /// Get the network-encoded fields length of the transaction.
+    pub fn network_encoded_fields_length(&self, signature: &Signature) -> usize {
```

Comment:
what's a usecase for this fn? I think this is same as `self.rlp_header_signed().payload_length`?

Reply:
removed in 72169eb5, as it was unused anyway